src/components/ha-sidebar.ts

```diff
@@ -335,6 +335,27 @@ class HaSidebar extends LitElement {
         // @ts-ignore
         selectedEl.scrollIntoViewIfNeeded();
       }
+      this.setTitle();
+    }
+  }
+
+  private setTitle() {
```

Comment:
This is not something that should be managed by the sidebar. This should be a new state mixin for hass-element.

We should also extract the getPanelTitle function and move it to `data/panel.ts`

Reply:
:ok_hand: